### ISSUE TITLE
MixpanelInstance.reset() is async and should have a completion once a new distinctId is ready.

### ISSUE THREAD

**Issue by marcelodeaguiar** [mixpanel/mixpanel-swift#465]
Calling reset on MixpanelInstance and then trying to get distinctId does not work since reset is async and add to trackingQueue. This method should have a completion once a new distinctId is ready to be used.

As a workaround I`m using 

```
Mixpanel.mainInstance().reset()
Mixpnael.mainInstance().flush() { /* use the new distinctId here */ }
```

**Comment by zihejia** [comment]
thanks @marcelodeaguiar for bringing this up. We will address this issue.

**Comment by zihejia** [comment]
Hi @marcelodeaguiar , the fix is available in 2.10.3.
